mergify_engine/context.py

```diff
@@ -136,7 +137,10 @@ def set_summary_check(
         self._save_cached_last_summary_head_sha(self.pull["head"]["sha"])
 
         try:
-            return check_api.set_check_run(self, self.SUMMARY_NAME, result)
+            check = check_api.set_check_run(self, self.SUMMARY_NAME, result)
+            if not check:
+                raise RuntimeError("setting summary check-run failed")
```

Comment:
This can't happen, I have added that because of mypy.

`set_check_run` can return `None` only if `update_only=True` is set.

Another way could be to have `set_check_run` and `update_check_run` instead of the `update_only` flag. Mypy will be happy too.

Reply:
This is not needed anymore.